src/components/job/UpsertJob.jsx

```diff
@@ -1,30 +1,78 @@
 import { useRequest } from "ahooks";
-import { React } from "react";
+import { React, useEffect, useMemo, useState } from "react";
 import { useForm } from "react-hook-form";
 import { toast } from "react-hot-toast";
-import { useNavigate } from "react-router-dom";
+import { useLocation, useNavigate, useParams } from "react-router-dom";
 
 import { ENGLISH_LEVEL, JOB_STATUS, SHIFT_TYPE, WAGE_TYPE } from "@/constants";
 import { useAuth, useJob } from "@/models";
 import { getCoordinate } from "@/utils";
 
-export default function AddJob() {
+import Center from "../Center";
+import Spin from "../Spin";
+const LOCAL_KEY = "REFUGEE_ONE_JOB_DRAFT";
+
+export default function UpsertJob({ update }) {
   const auth = useAuth();
   const job = useJob();
   const navigate = useNavigate();
+  const { jobId } = useParams();
+  const location = useLocation();
+  const from = location.state?.from || "/";
+
+  const {
+    run: getJob,
+    data: jobData,
+    loading: preparing,
+  } = useRequest(async (jobId) => job.getJob(jobId), {
+    manual: true,
+    onSuccess: (data) => {
+      data.dateJobStart = data.dateJobStart.toDate().toISOString();
+      loadData(data);
+    },
+    onError: (error) => {
+      console.error(error);
+      toast.error("Failed to get job information");
+    },
+  });
+
+  useEffect(() => {
+    if (update) {
+      (async () => getJob(jobId))();
+    }
+  }, []);
+
+  // in update mode, we do not enable load draft feature
+  const [showModal, setShowModal] = useState(
+    !update && window.localStorage.getItem(LOCAL_KEY) !== null
+  );
+
   const { register, handleSubmit, setValue, getValues } = useForm({
+    // if jobData has a value, it means we are editing the job
     // if the user is an employer, has a company, autofill the company name
-    defaultValues: { company: auth.user?.company || "" },
+    defaultValues: jobData || {
+      company: auth.user?.company || "",
+      dateJobStart: Date.now(),
+    },
   });
 
-  const { run: createJob, loading } = useRequest(
+  const { run: upsertJob, loading } = useRequest(
     async (data) => {
+      data.dateJobStart = new Date(data.dateJobStart);
       const coordinate = await getCoordinate(
         data.address.street,
         data.address.city,
         data.address.state,
         data.address.zipcode
       );
+      // edit
+      if (jobData) {
+        Object.entries(data).forEach(([key, value]) => {
+          jobData[key] = value;
+        });
+        return job.updateJob(jobId, jobData);
+      }
+      // create
       return job.createJob({
         ...data,
         adminMessage: "",
```

Comment:
Location field (string concat from address field) is not updated, so job view still showing the old data.